docs/sql/statements/insert.rst

```diff
@@ -26,130 +31,155 @@ Synopsis
 
-        select unnest([{foo=2}, {foo='a string'}])
+    column_ident [, ... ]
 
-    or::
+And ``assignments`` can be one or more column assignments::
 
-        select unnest([{_invalid_col='foo', valid_col='bar'}])
+    column_ident = expression [, ... ]
 
-  * If you select from an ignored object which contains different data
-    types for the same object column, e.g.::
+Within the ``assignments``, you can use a virtual table named ``excluded`` to
+reference values from the ``INSERT`` columns.  For example::
 
-        insert into from_table (o) values ({col='foo'}),({col=1})
-        insert into to_table (i) (select o['col'] from t)
+     INSERT INTO my_table (col_a, col_b) VALUES (1, 41)
+     ON CONFLICT (col_a) DO UPDATE SET col_b = excluded.col_b + 1
 
-Any updates which happened before the failure will be persisted, which will
-lead to inconsistent outcomes. So special care needs to be taken by the
-application when using statements which might produce dynamic data.
+The above statement would update ``col_b`` to ``42`` if ``col_a`` was a primary
+key and the value ``1`` already existed for ``col_a``.
 
-Parameters
-==========
 
-:table_ident:
-    The identifier (optionally schema-qualified) of an existing table.
+.. _sql-insert-on-conflict-do-nothing:
 
-:column_ident:
-    The name of a column or field in the table pointed to by *table_ident*.
+``ON CONFLICT DO NOTHING``
+--------------------------
 
-:expression:
-    An :ref:`expression <gloss-expression>` or value to assign to the
-    corresponding column.
+If you use the ``ON CONFLICT DO NOTHING`` clause, CrateDB will silently ignore
+rows that would cause a duplicate key conflict (i.e., CrateDB will not insert
+them and will not produce an error). For example::
 
-:query:
-    A query (``SELECT`` statement) that supplies the rows to be inserted.
-    Refer to the ``SELECT`` statement for a description of the syntax.
+     INSERT INTO my_table (col_a, col_b) VALUES (1, 42)
+     ON CONFLICT DO NOTHING
 
-:output_expression:
-    An expression to be computed and returned by the ``INSERT`` command
-    after each row is updated. The expression can use any column names
-    of the table or use ``*`` to return all columns. :ref:`System columns
-    <sql_administration_system_columns>` can also be returned.
+In the statement above, if ``col_a`` had a primary key constraint and the value
+``1`` already existed for ``col_a``, CrateDB would not perform an insert.
 
-:output_name:
-    A name to use for the result of the output expression.
+You may limit the scope of this clause by specifying one or more of the primary
+key columns after ``ON CONFLICT``, like so::
+
+     INSERT INTO my_table (col_a, col_a) VALUES (1, 42)
+     ON CONFLICT (col_a) DO NOTHING
+
+In this example, CrateDB would only silently ignore conflicting inserts for
+``col_a``. CrateDB would not ignore conflicting values for other primary keys
+(if they exist).
```

Comment:
Did you test if this actually works? I'm not sure if we support this.

(Please push a dedicated fixup commit next time instead of squashing,  - makes it easier to review the follow up changes)

Reply:
@norosa could you get back to this so we can wrap this up?